Tasks/DeployAzureResourceGroup/VirtualMachine.ts

```diff
@@ -79,11 +82,17 @@ export class VirtualMachine {
                     }
                     break;
                 case "Delete":
-                    for (var i = 0; i < listOfVms.length; i++) {
-                        var vmName = listOfVms[i]["name"];
-                        console.log(tl.loc("VM_Delete", vmName));
-                        client.virtualMachines.deleteMethod(this.taskParameters.resourceGroupName, vmName, this.postOperationCallBack);
+                    var extDelPromise = this.machineGroupAgentExtensionManager.deleteMGExtension();
```

Comment:
Changed. Added a separate method for deleting vms. Deleting RG still uses the old method.